brave/brave-core
# Fixes wallet bad state

Resolves https://github.com/brave/brave-browser/issues/3061
Resolves https://github.com/brave/brave-browser/issues/3060

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:

1. go to master
2. clean profile
3. enable rewards
4. save recovery keys
5. close browser
6. open ledger_state and save addresses
7. rename addresses to something else
8. change seed to `OIeFXZsZd5rHDzAMwN1I0F1AX4S9KPM78uNTvmJZak8=`
9. start browser again
10. make sure that addresses were not updated and see that recovery keys are still the same
11. switch to this PR's branch
12. start the browser
13. check recovery keys, make sure that they are `decade destroy first curtain number sniff bring toward art alone fall pass staff cool base pink veteran test rhythm polar tower coach powder wave`
14. close browser
15. open ledger_state and make sure that addresses are the same as they were in step 6

## Reviewer Checklist:
